app_integrations/apps/salesforce.py

```diff
@@ -0,0 +1,395 @@
+                                'characters'),
+                'format': re.compile(r'^[a-zA-Z0-9._#@]{85}$')
+            },
+            'client_secret': {
+                'description': ('The consumer secret from the Salesforce connected App. '
+                                'This should be a string of 19 numeric characters'),
+                'format': re.compile(r'^[0-9]{19}$')
+            },
+            'username': {
+                'description': ('The username of a user account. This should be an '
+                                'Email address'),
+                'format': re.compile(r'^[A-Za-z0-9-_.+]+@[A-Za-z0-9-.]+\.[A-Za-z]{2,}$')
+            },
+            'password': {
+                'description': ('The password of a user account. This should be a '
+                                'random string'),
+                'format': re.compile(r'.*')
+            },
+            'security_token': {
+                'description': ('The security token generated in user account. This '
+                                'should be a string of 24 alphanumeric characters'),
+                'format': re.compile(r'^[a-zA-Z0-9]{24}$')
+            }
+        }
+
+    def _validate_status_code(self, resp):
+        """Validate status code from get response
+
+        Args:
+            resp (Response): The response object from get request
+
+        Returns:
+            bool: Returns True if status_code is 200. Otherwise returns False,
+                except
+        """
+        if resp.status_code == 200:
+            return True
+        if resp.status_code == 401:
+            # The OAuth token used has expired or is invalid.
+            # Retry to renew OAuth token
+            LOGGER.error('The OAuth token used has expired or is invalid. '
+                         'Error code %s, error message %s',
+                         resp.json().get('errorCode', None), resp.json().get('message', None))
+            self._request_token()
+
+            # Get request will retry when SalesforceAppError exception raised
+            raise SalesforceAppError
+        elif resp.status_code == 403 and resp.json().get('errorCode') == 'REQUEST_LIMIT_EXCEEDED':
+            # Exceeded API request limits in your org. Log this information for
+            # future reference.
+            LOGGER.error('Exceeded API request limits')
+            return False
+        elif resp.status_code == 500:
+            # Server internal error. Get request will retry.
+            raise SalesforceAppError
+        elif resp.status_code > 200:
+            LOGGER.error('Unexpected status code %d detected, error message %s',
+                         resp.status_code,
+                         resp.json())
+            return False
+
+    def _make_get_request(self, full_url, headers, params=None):
+        """Make GET request with backoff logic
+
+        This method overrides the method in AppIntegration base class to handle
```

Comment:
I think we can probably modify the base class's method to handle returning data as json or raw response object. you could still have something in here for doing the backoff, but ideally all get requests would go through the base class if possible. we can chat more about how to achieve this (i'm thinking just have a `kwargs` argument that could include a `return_json` param or something)

Reply:
this is great added context - thank you for clarifying!! in this case, I think you can leave it as is. we can investigate ways to make the subclass more flexible/adaptable in the future :)